src/extra/gpkg.jl

```diff
@@ -0,0 +1,1054 @@
+  table::T
+  crs::Any
+end
+
+# Implement Tables.jl interface for the wrapper
+Tables.istable(::Type{GeoPackageTable{T}}) where T = Tables.istable(T)
+Tables.rowaccess(::Type{GeoPackageTable{T}}) where T = Tables.rowaccess(T)
+Tables.columnaccess(::Type{GeoPackageTable{T}}) where T = Tables.columnaccess(T)
+Tables.rows(gt::GeoPackageTable) = Tables.rows(gt.table)
+Tables.columns(gt::GeoPackageTable) = Tables.columns(gt.table)
+Tables.columnnames(gt::GeoPackageTable) = Tables.columnnames(gt.table)
+Tables.getcolumn(gt::GeoPackageTable, i::Int) = Tables.getcolumn(gt.table, i)
+Tables.getcolumn(gt::GeoPackageTable, nm::Symbol) = Tables.getcolumn(gt.table, nm)
+Tables.schema(gt::GeoPackageTable) = Tables.schema(gt.table)
+
+# Implement GeoInterface.crs method
+GeoInterface.crs(gt::GeoPackageTable) = gt.crs
+
+const WKB_POINT = 0x00000001
+const WKB_LINESTRING = 0x00000002
+const WKB_POLYGON = 0x00000003
+const WKB_MULTIPOINT = 0x00000004
+const WKB_MULTILINESTRING = 0x00000005
+const WKB_MULTIPOLYGON = 0x00000006
+const WKB_GEOMETRYCOLLECTION = 0x00000007
+
+const WKB_Z = 0x80000000
+const WKB_M = 0x40000000
+const WKB_ZM = 0xC0000000
+
+function parse_gpb_header(blob::Vector{UInt8})
+  io = IOBuffer(blob)
+  
+  magic = read(io, UInt16)
+  magic == 0x5047 || error("Invalid GeoPackage binary header")
+  
+  version = read(io, UInt8)
+  flags = read(io, UInt8)
+  
+  srs_id = ltoh(read(io, Int32))
+  
+  envelope_type = flags >> 1 & 0x07
+  has_envelope = envelope_type > 0
+  
+  min_x = min_y = max_x = max_y = nothing
+  min_z = max_z = min_m = max_m = nothing
+  
+  if has_envelope
+    if envelope_type >= 1
+      min_x = ltoh(read(io, Float64))
+      max_x = ltoh(read(io, Float64))
+      min_y = ltoh(read(io, Float64))
+      max_y = ltoh(read(io, Float64))
+    end
+    if envelope_type >= 2
+      min_z = ltoh(read(io, Float64))
+      max_z = ltoh(read(io, Float64))
+    end
+    if envelope_type >= 3
+      min_m = ltoh(read(io, Float64))
+      max_m = ltoh(read(io, Float64))
+    end
+  end
+  
+  header = GeoPackageBinaryHeader(
+    magic, version, flags, srs_id,
+    min_x, max_x, min_y, max_y,
+    min_z, max_z, min_m, max_m
+  )
+  
+  return header, position(io)
+end
+
+function parse_wkb_point(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool, crs_type)
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  
+  x = read_value(read(io, Float64))
+  y = read_value(read(io, Float64))
+  
+  if isnan(x) && isnan(y)
+    return nothing
+  end
+  
+  coords_list = [x, y]
+  
+  if has_z
+    z = read_value(read(io, Float64))
+    push!(coords_list, z)
+  end
+  
+  if has_m
+    m = read_value(read(io, Float64))
+  end
+  
+  # Create point with appropriate CRS
+  if crs_type <: LatLon
+    if length(coords_list) == 2
+      return Point(crs_type(y, x))  # LatLon(lat, lon) - note swapped order
+    else
+      return Point(crs_type(y, x, coords_list[3]))
+    end
+  else
+    return Point(coords_list...)
+  end
```

Comment:
We need to review this section. There are various CRS types defined in CoordRefSystems.jl, including `LatLon` and `LatLonAlt` (the first two branches in your code) and various projected systems like `Albers`, `Mercator`, etc. Your third branch is assuming a `Cartesian` type. You probably need read the `srs_id` from the header to determine which type is the correct one.